docs/pubsub-api.rst

```diff
@@ -0,0 +1,255 @@
+``gcloud.pubsub`` API
+=====================
+
+Connection / Authorization
+--------------------------
+
+- Inferred defaults used to create connection if none configured explicitly:
+
+  - credentials (derived from GAE / GCE environ if present).
+
+  - ``project_id`` (derived from GAE / GCE environ if present).
+
+  - ``scopes``
+
+
+Manage topics for a project
+---------------------------
+
+Create a new topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.create_topic("topic_name")
+   >>> topic.name
+   'topic_name'
+
+Create a new topic for an explicit project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.create_topic("topic_name", project_id="my.project")
+   >>> topic.name
+   'topic_name'
+
+Fetch an extant topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.name
+   'topic_name'
+
+Fetch an extant topic for the default project:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name', project_id="my.project")
+   >>> topic.name
+   'topic_name'
+
+Attempt to fetch a non-extant topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> pubsub.get_topic('nonesuch')
```

Comment:
I think it's problematic if you issue a Cloud Pub/Sub API call when you do:
```python
pubsub.get_topic('nonesuch')
```
because it forces users to issue 2 requests for publishing/pulling or whatever. The first 'get' request is totally unnecessary.